internal/tasks/manifests.go

```diff
@@ -598,6 +605,37 @@ func (j *Janitor) doVulnerabilityCheck(account keppel.Account, repo keppel.Repos
 			//after successful replication, restart this call to read the new blob with the correct StorageID from the DB
 			return j.doVulnerabilityCheck(account, repo, manifest)
 		}
+
+		if blob.BlocksVulnScanning == nil && strings.HasSuffix(blob.MediaType, "gzip") {
+			reader, _, err := j.sd.ReadBlob(account, blob.StorageID)
+			if err != nil {
+				return err
+			}
+			defer reader.Close()
+			gzipReader, err := gzip.NewReader(reader)
+			if err != nil {
+				return err
+			}
+			defer gzipReader.Close()
+
+			numberBytes, err := io.Copy(io.Discard, gzipReader)
+			if err != nil {
+				return err
+			}
+
+			b2p := func(b bool) *bool {
+				return &b
+			}
+			// mark blocked for vulnerability scanning if one layer/blob is bigger than 10 GiB
+			blob.BlocksVulnScanning = b2p(numberBytes >= int64(1<<30*blobUncompressedSizeToBigGib))
```

Comment:
Since `b2p` is only once, we can do without an anon function.
```suggestion
			// mark blocked for vulnerability scanning if one layer/blob is bigger than 10 GiB
			blocksVulnScanning := b2p(numberBytes >= int64(1<<30*blobUncompressedSizeToBigGib))
			blob.BlocksVulnScanning = &blocksVulnScanning
```